main.nf

```diff
@@ -1578,7 +1579,7 @@ process circularmapper{
     params.mapper == 'circularmapper'
 
     script:
-    def filter = params.circularfilter ? '' : '-f true -x false'
+    def filter = params.circularfilter ? '-f true -x false' : ''
```

Comment:
Would it be better to explicitly state the options given when `params.circularfilter` is not provided?
I believe they are `-f false -x false`, which then begs the question why `-x false` is given at all since it is the default?

Reply:
> usage: RealignSAMFilev1.0
>  -f,--filterCircularReads <FILTER>                     filter the reads
>                                                        that don't map to a
>                                                        circular identifier
>                                                        (true/false),
>                                                        **default false**
> [...]
>  -x,--filterNonCircularSequences <FILTERSEQUENCEIDS>   filter the sequence
>                                                        identifiers that
>                                                        are not circular
>                                                        identifiers
>                                                        (true/false),
>                                                        **default false**

Reply:
Not 100% sure what you mean - if `params.circularfilter` is not supplied, there are no extra parameters?

@apeltzer will need to say with `-x` is supplied...

Reply:
Ah maybe `-x` is for the BAM header?

Reply:
One of the specified parameters seems to reinforce the default, which is a bit confusing. So if providing `-x false` is not needed, then it should not be included in the definition. If it is, then the defaults of both should be given instead of `''`.
It is not that important in terms of the code, but in terms of consistency. I don't see why on one side we _specify_ one of the defaults, but on the other we don't provide it _because_ it is the default.

Reply:
Ah Ok. Yeah I dunno, need @apeltzer 's input for design reasons here... never really played with CM.

Reply:
That bit was implemented by Stephen tbf ;-) Have to check, never really used this myself.

Reply:
See email from Stephen. Updated accoridngly so bot hget filtered out!